Extensions/render/medium.lisp

```diff
@@ -187,12 +232,32 @@
 				(+ (floor text-height 2))))
 		    (:baseline y)
 		    (:bottom (+ y (- baseline text-height))))))
-	(multiple-value-bind (x y)
-	    (transform-position (sheet-native-transformation
-				   (medium-sheet medium))
-				x y)
-	  (let ((paths (string-primitive-paths x y string xfont size)))
-	    (%medium-fill-paths medium paths nil)))))))
+	(let ((paths (string-primitive-paths x y string xfont size
+					     (lambda (paths opacity-image dx dy transformation)
+					       (let ((msheet (sheet-mirrored-ancestor (medium-sheet medium))))
```

Comment:
for such big function I think that it would be good to factor it into toplevel helper function